### PR TITLE
chore : added eslint prettier with webpack-defaults standard

### PR DESCRIPTION
**Pull request by anikethsaha**
This PR contains
- eslint config
- prettier config
- linting and pretesting scripts
- lockfiles update (opt-in-out)
- formatting and linting fixes to the `src` and `test` folder

### CODE REVIEW

**Comment by anikethsaha** [comment]
Cc @evilebottnawi .

File: package.json [hunk 1]
```diff
@@ -37,6 +42,7 @@
   },
   "homepage": "https://github.com/ericclemmons/npm-install-webpack-plugin#readme",
   "dependencies": {
+    "@webpack-contrib/eslint-config-webpack": "^3.0.0",
```
Comment (evilebottnawi) [inline package.json]: Move this to `devDependencies`

**Comment by anikethsaha** [comment]
cc @evilebottnawi

**Comment by evilebottnawi** [comment]
Can we fix CI problems?

**Comment by anikethsaha** [comment]
the CI issue is being handled in https://github.com/webpack-contrib/npm-install-webpack-plugin/pull/130

Actually the changes need to be done here https://github.com/webpack-contrib/npm-install-webpack-plugin/blob/master/test/plugin.test.js#L15-L45

the mock compiler is of v3, I need to change it to v4. Which I am doing in https://github.com/webpack-contrib/npm-install-webpack-plugin/pull/130

**Comment by anikethsaha** [comment]
Else, the linting+prettier tests are all fine, just few linting warnings.

**Pull request merged**